addon_service/management/commands/fix_migrating_users_with_mutiple_external_accounts.py

```diff
@@ -0,0 +1,271 @@
+
+    return configured_addon
+
+
+def get_external_service(provider):
+    return ExternalService.objects.get(wb_key=provider)
+
+
+def get_user_reference(user_guid):
+    return UserReference.objects.get_or_create(user_uri=f"{OSF_BASE}/{user_guid}")
+
+
+def get_or_create_authorized_account(external_account, provider, user_guid):
+    if provider in storage_provider:
+        AuthorizedAccount = AuthorizedStorageAccount
+    elif provider in citation_provider:
+        AuthorizedAccount = AuthorizedCitationAccount
+
+    external_service = get_external_service(provider)
+    account_owner = get_user_reference(user_guid)
+
+    authorized_account = None
+    if not AuthorizedAccount.objects.filter(
+        external_account_id=external_account.id,
+        external_service=external_service,
+        account_owner=account_owner,
+    ).exists():
+        credentials = get_credentials(external_service, external_account)
+        authorized_account = AuthorizedAccount(
+            display_name=external_service.display_name.capitalize(),
+            int_authorized_capabilities=(
+                AddonCapabilities.UPDATE | AddonCapabilities.ACCESS
+            ).value,
+            account_owner=account_owner,
+            external_service=external_service,
+            credentials=credentials,
+            external_account_id=external_account.provider_id,
+        )
+    else:
+        authorized_account = AuthorizedAccount.objects.get(
+            external_account_id=external_account.id,
+            external_service=external_service,
+            account_owner=account_owner,
+        )
+
+    return authorized_account
+
+
+# check to see if the ConfiguredAddon's base account has the same
+# external_account_id as the ExternalAccount's provider_id
+def configured_addon_has_correct_base_account(external_account, node_guid, provider):
+    configured_addon = get_configured_addon(node_guid, provider)
+    if (
+        configured_addon.base_account.external_account_id
+        == external_account.provider_id
+    ):
+        return True
+    return False
+
+
+def fix_migration_for_user_and_provider(user_guid, provider):
+    if not user_guid:
+        return
+
+    for node_settings in get_node_settings_for_user_and_provider(user_guid, provider):
+        external_account = node_settings.external_account
+        node_guid = get_node_guid(node_settings.owner_id)
+
+        # check to see if the migrated configured_addon instance have the authorized account with the same external_account_id
+        if not configured_addon_has_correct_base_account(
+            external_account, node_guid, provider
+        ):
+            # if not, then we fix it
+            configured_addon = get_configured_addon(node_guid, provider)
+            authorized_account = get_or_create_authorized_account(
+                external_account, provider, user_guid
+            )
+            configured_addon.base_account = authorized_account
```

Comment:
Could you please add some logging in here as well, to track what's being wired up?